python/lsst/ap/pipe/createApFakes.py

```diff
@@ -0,0 +1,300 @@
+
+        # Compute the rotation matrix to move our random points to the
+        # correct location.
+        rotMatrix = self._createRotMatrix(boundingCircle)
+        randVect = np.dot(rotMatrix, randVect.transpose()).transpose()
+        decs = np.arcsin(randVect[:, 2])
+        ras = np.arctan2(randVect[:, 1], randVect[:, 0])
+
+        return {self.config.decColName: decs,
+                self.config.raColName: ras}
+
+    def _createRotMatrix(self, boundingCircle):
+        """Compute the 3d rotation matrix to rotate the dec=90 pole to the
+        center of the circle bound.
+
+        Parameters
+        ----------
+        boundingCircle : `lsst.sphgeom.BoundingCircle`
+             Circle bound covering the tract.
+
+        Returns
+        -------
+        rotMatrix : `numpy.ndarray`, (3, 3)
+            3x3 rotation matrix to rotate the dec=90 pole to the location of
+            the circle bound.
+        """
+        center = boundingCircle.getCenter()
+        cross = np.array([-center.y(),
+                          center.x(),
+                          0])
+        cross /= np.sqrt(cross[0] ** 2 + cross[1] ** 2 + cross[2] ** 2)
+        cosTheta = center.z()
+        sinTheta = np.sin(np.arccos(center.z()))
+
+        rotMatrix = cosTheta * np.array([[1, 0, 0],
+                                         [0, 1, 0],
+                                         [0, 0, 1]])
+        rotMatrix += sinTheta * np.array([[0, -cross[2], cross[1]],
+                                          [cross[2], 0, -cross[0]],
+                                          [-cross[1], cross[0], 0]])
+        rotMatrix += (
+            (1 - cosTheta)
+            * np.array(
+                [[cross[0] ** 2, cross[0] * cross[1], cross[0] * cross[2]],
+                 [cross[0] * cross[1], cross[1] ** 2, cross[1] * cross[2]],
+                 [cross[0] * cross[2], cross[1] * cross[2], cross[2] ** 2]])
+        )
+        return rotMatrix
+
+    def createVisitCoaddSubdivision(self, nFakes):
+        """Assign a given fake either a visit image or coadd based on the
+        ``faction`` config value.
+
+        Parameters
+        ----------
+        nFakes : `int`
+            Number of fakes to create.
+
+        Returns
+        -------
+        output : `dict`
+        """
+        nBoth = int(self.config.fraction * nFakes)
+        nOnly = int((1 - self.config.fraction) / 2 * nFakes)
+        isVisitSource = np.zeros(nFakes, dtype=bool)
+        isTemplateSource = np.zeros(nFakes, dtype=bool)
+        if nBoth > 0:
+            isVisitSource[:nBoth] = True
+            isTemplateSource[:nBoth] = True
+        if nOnly > 0:
+            isVisitSource[nBoth:(nBoth + nOnly)] = True
+            isTemplateSource[(nBoth + nOnly):] = True
+
+        return {self.config.visitSourceFlagCol: isVisitSource,
+                self.config.templateSourceFlagCol: isTemplateSource}
+
+    def createRandomMagnitudes(self, nFakes):
```

Comment:
Please emhpasize here (`createUniformRandomMagnitudes()`) and note in the Task docstring that this is _uniform in magnitudes_ which is a non-physical model but perhaps all right for this task.

Reply:
I kind of want to leave this as a generic name so that it can be overwritten in possible inherited tasks.